src/controller/auth.ts

```diff
@@ -0,0 +1,86 @@
+import { Request , Response, NextFunction } from "express";
+import bcrypt from 'bcryptjs'
+import jwt from 'jsonwebtoken'
+import Recruiter from '../models/recruiter';
+import { validationResult } from 'express-validator';
+import dotenv from 'dotenv';
+
+dotenv.config();
+
+export const addRecruiter = (req: Request, res: Response, next: NextFunction) => {
+    const name = req.body.name;
+    const email = req.body.email;
+    const password = req.body.password;
+    const company = req.body.company;
+    const phone = req.body.phone;
+
+    const valError = validationResult(req);
+    if(!valError.isEmpty()){
+        const error = new Error('Validation failed.');
+        error.statusCode = 422;
+        error.data = valError.array();
+        throw error;  
+      }
+
+    bcrypt.hash(password,12)
+    .then(hashedPassword => {
+        const recruiter = new Recruiter({
+            name: name,
+            email: email,
+            password: hashedPassword,
+            company:company,
+            phone: phone
+        });
```

Comment:
you can update password before passing the object and instead of passing the individual property pass req.body
req.body.password = hashedPassword;
new Recruiter(req.body)